src/connection.js

```diff
@@ -0,0 +1,272 @@
+/*
+ * Copyright 2022, Tony Atkins
+ * Copyright 2011-2020, Colin Clark
+ *
+ *  Licensed under the MIT license, see LICENSE for details.
+ */
+
+(function (fluid) {
+    "use strict";
+    var youme = fluid.registerNamespace("youme");
+
+    /**
+     *
+     * A MIDI Connection represents a connection to a single input or output port.  Unless you are completing the wiring
+     * yourself, you should not use this directly, but should use `youme.connection.input` or `youme.connection.output`
+     * instead.
+     *
+     **/
+    fluid.defaults("youme.connection", {
+        gradeNames: ["fluid.component"],
+
+        openImmediately: false,
+
+        portSpec: {},
+
+        members: {
+            port: false
+        },
+
+        events: {
+            onPortOpen: null,
```

Comment:
The port may be repeatedly opened and closed, but I think it might be useful for the `onReady` event in `youme.system`.  Are there examples of this type of usage I could refer to?

Reply:
I tried this for a while and couldn't get the resource loading to block `onCreate`.  [Here's a pen](https://codepen.io/duhrer/pen/BaryBQv) with an example, in all cases `onCreate` fires before the promise is resolved.

Reply:
I updated the example, it seems like onCreate does wait for the promise to resolve, but always beats anything else "thenned" to the same promise.

Reply:
Ah, I see, it's down to the order in which they were registered.  I will give that some thought and work on the timing tomorrow.